### PR TITLE
Prevent NPEs in math:* functions. Tested in XQTS suite

### PR DESCRIPTION
**Pull request by dizzzz**
NPEs found in XQTS suite for math functions. 

image

one issue remaining, see below

### CODE REVIEW

**Comment by dizzzz** [comment]
one issue remaining in xqts: 
image

**Comment by dizzzz** [comment]
Oops, c73f6d4ff9509bc3caa65cd2f954828e0b9b7a78 and b53fd085077b3db67f4236a21ebb371fc8420e7c  need to be removed here

**Comment by dizzzz** [comment]
ready for review

File: exist-core/src/main/java/org/exist/xquery/functions/math/NoParamFunctions.java [hunk 1]
```diff
@@ -65,15 +64,6 @@ public int getDependencies() {
          */
     public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
 
-    	if (context.getProfiler().isEnabled()) {
```
Comment (adamretter) [inline exist-core/src/main/java/org/exist/xquery/functions/math/NoParamFunctions.java:68]: Did you mean to remove this?

**Comment by adamretter** [comment]
@dizzzz You might have to talk me through it as it looks like code-reformatting got mixed into the NPE commit, and now "I can't see the wood for the trees".

**Comment by dizzzz** [comment]
craps I did not see the whitespace change; This one is better readable: https://github.com/eXist-db/exist/pull/3691/files?diff=split&w=1

**Comment by dizzzz** [comment]
this is the GitHub trick: 
image

File: exist-core/src/main/java/org/exist/xquery/functions/math/TwoParamFunctions.java [hunk 2]
```diff
@@ -24,100 +24,75 @@
...
+            new QName(POW, MathModule.NAMESPACE_URI, MathModule.PREFIX),
+            "Returns the result of raising the first argument to the power of the second.",
+            new SequenceType[]{
+                    new FunctionParameterSequenceType("value", Type.DOUBLE, Cardinality.ZERO_OR_ONE, "The value"),
+                    new FunctionParameterSequenceType("power", Type.NUMBER, Cardinality.EXACTLY_ONE, "The power to raise the value to")
+            },
+            new FunctionReturnSequenceType(Type.DOUBLE, Cardinality.ZERO_OR_ONE, "the result")
     );
+    @SuppressWarnings("unused")
+    private static final Logger logger = LogManager.getLogger(TwoParamFunctions.class);
```
Comment (adamretter) [inline exist-core/src/main/java/org/exist/xquery/functions/math/TwoParamFunctions.java:61]: If its dead-code, please just remove it.

File: exist-core/src/main/java/org/exist/xquery/functions/math/NoParamFunctions.java (hunk 1 above)
Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/math/NoParamFunctions.java:68]: yes, as BasicFunction implements this functionality already.

**Comment by dizzzz** [comment]
Good to go....

**Comment by joewiz** [comment]
@dizzzz To confirm, is the "one issue remaining in xqts" now fixed too?

**Comment by dizzzz** [comment]
@joewiz No the 'one remaining issue' shown in im image above, is not in the math library. I will not fix this, sorry.

**Pull request merged**